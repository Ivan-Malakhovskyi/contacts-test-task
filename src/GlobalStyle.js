import { createGlobalStyle } from "styled-components";


 export const theme = {
     colors: {
     MainBackgroundColor: '#ffffff',
       addBgColor: '#EDEDED',
    MainColorInput: '#AAAAAA'
    }
}

export const GlobalStyle = createGlobalStyle`


body {
  background-color: ${({theme: {colors}}) => colors.MainBackgroundColor};
  font-family: "Poppins", sans-serif;
  font-size: 16px;
    font-weight: 400;
  font-style: normal;
  line-height: 1.5;
}

img {
  display: block;
  max-width: 100%;
  object-fit: cover; 
  height: auto;
  margin: 0 auto;
}

ul,li,h1, h2, h3, h4, h5, h6, p {
font-family: inherit;
padding: 0;
  margin: 0;
  list-style: none;
}

a {
  font-family: inherit;
  text-decoration: none;
  display: inline-block;
}

button {
  font-family: inherit;
  cursor: pointer;
  border: none;
  background-color: transparent;
}

`