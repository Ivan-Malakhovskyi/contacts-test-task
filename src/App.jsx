import { Route, Routes } from "react-router-dom";
import "./App.css";
import { GlobalStyle } from "./GlobalStyle";
import { Home } from "./components/pages/Home";
import { ContactDetails } from "./components/pages/ContactDetails";
import { Layout } from "./components/layout/Layout";

const App = () => {
  return (
    <>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="/contact/:id" element={<ContactDetails />} />
          <Route path="*" element={<p>Not found</p>} />
        </Route>
      </Routes>

      <button
        type="button"
        onClick={() => {
          console.log("Click");
        }}
      >
        Click
      </button>

      <GlobalStyle />
    </>
  );
};

export default App;
