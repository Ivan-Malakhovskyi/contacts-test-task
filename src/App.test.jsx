import { render, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
import App from "./App";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "./components/redux/store";
import { ThemeProvider } from "styled-components";
import { theme } from "./GlobalStyle";
import "@testing-library/jest-dom";
import { act } from "react";

test("render Home component at root path", () => {
  act(() => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/"]}>
          <ThemeProvider theme={theme}>
            <App />
          </ThemeProvider>
        </MemoryRouter>
      </Provider>
    );
  });

  const createContactElement = screen.getByText("Contacts");
  expect(createContactElement).toBeInTheDocument();
});

it("when btn clicked, cb should be called", () => {
  const spy = jest.fn();
  act(() => {
    render(<button onClick={spy}>Click</button>);
  });

  screen.getByText("Click").click();
  expect(spy).toHaveBeenCalled();
});

it("snapshot", () => {
  const tree = create(<button onClick={() => null}>Click</button>).toJSON();

  expect(tree).toMatchSnapshot();
});
