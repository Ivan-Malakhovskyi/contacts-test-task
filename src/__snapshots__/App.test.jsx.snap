// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<button
  onClick={[Function]}
>
  Click
</button>
`;
